UnitTests/ViewTests.cs

```diff
@@ -543,5 +543,173 @@ public void CanFocus_Set_Changes_TabIndex_And_TabStop ()
+				Assert.Equal (0, sv1.Frame.Width);
+				Assert.Equal (0, sv1.Frame.Height);
+			};
+
+			t.Load += () => {
+				tc++;
+				Assert.Equal (1, tc);
+				Assert.Equal (1, rc);
+				Assert.Equal (1, v1c);
+				Assert.Equal (1, v2c);
+				Assert.Equal (1, sv1c);
+
+				Assert.True (t.CanFocus);
+				Assert.False (r.CanFocus);
+				Assert.False (v1.CanFocus);
+				Assert.False (v2.CanFocus);
+				Assert.True (sv1.CanFocus);
+
+				Application.Refresh ();
+			};
+			t.Initialized += () => {
+				rc++;
+				Assert.Equal (t.Frame.Width, r.Frame.Width);
+				Assert.Equal (t.Frame.Height, r.Frame.Height);
+				v1c++;
+				Assert.Equal (t.Frame.Width, v1.Frame.Width);
+				Assert.Equal (t.Frame.Height, v1.Frame.Height);
+				v2c++;
+				Assert.Equal (t.Frame.Width, v2.Frame.Width);
+				Assert.Equal (t.Frame.Height, v2.Frame.Height);
+				sv1c++;
+				Assert.Equal (t.Frame.Width, sv1.Frame.Width);
+				Assert.Equal (t.Frame.Height, sv1.Frame.Height);
+				Assert.False (sv1.CanFocus);
+				sv1.CanFocus = true;
+				Assert.True (sv1.CanFocus);
+			};
+
+			v1.Add (sv1);
+			r.Add (v1, v2);
+			t.Add (r);
+
+			Application.Iteration = () => {
+				Application.Refresh ();
+				t.Running = false;
+			};
+
+			Application.Run (t, true);
+			Application.Shutdown (true);
+
+			Assert.Equal (1, tc);
+			Assert.Equal (1, rc);
+			Assert.Equal (1, v1c);
+			Assert.Equal (1, v2c);
+			Assert.Equal (1, sv1c);
+
+			Assert.True (t.CanFocus);
+			Assert.False (r.CanFocus);
+			Assert.False (v1.CanFocus);
+			Assert.False (v2.CanFocus);
+			Assert.True (sv1.CanFocus);
+		}
+
+		[Fact]
+		public void Initialized_Event_Will_Be_Invoked_Before_Redraw_Then_Added_Dynamically ()
+		{
+			Application.Init (new FakeDriver (), new NetMainLoop (() => FakeConsole.ReadKey (true)));
+
+			var t = new Toplevel ();
+
+			var r = new View ("r") { Width = Dim.Fill (), Height = Dim.Fill () };
+			var v1 = new View ("v1") { Width = Dim.Fill (), Height = Dim.Fill () };
+			var v2 = new View ("v2") { Width = Dim.Fill (), Height = Dim.Fill () };
+
+			int tc = 0, rc = 0, v1c = 0, v2c = 0, sv1c = 0;
+
+			t.Initialized += () => {
+				rc++;
+				Assert.Equal (t.Frame.Width, r.Frame.Width);
+				Assert.Equal (t.Frame.Height, r.Frame.Height);
+				v1c++;
+				Assert.Equal (t.Frame.Width, v1.Frame.Width);
+				Assert.Equal (t.Frame.Height, v1.Frame.Height);
+				v2c++;
+				Assert.Equal (t.Frame.Width, v2.Frame.Width);
+				Assert.Equal (t.Frame.Height, v2.Frame.Height);
+			};
+			t.Load += () => {
+				tc++;
+				Assert.Equal (1, tc);
+				Assert.Equal (1, rc);
+				Assert.Equal (1, v1c);
+				Assert.Equal (1, v2c);
+				Assert.Equal (0, sv1c);
+
+				Assert.True (t.CanFocus);
+				Assert.False (r.CanFocus);
+				Assert.False (v1.CanFocus);
+				Assert.False (v2.CanFocus);
+
+				Application.Refresh ();
+			};
+
+			r.Add (v1, v2);
+			t.Add (r);
+
+			Application.Iteration = () => {
+				var sv1 = new View ("sv1") { Width = Dim.Fill (), Height = Dim.Fill () };
+
+				sv1.Added += (e) => {
```

Comment:
I may be missing something, but I don't see where you test that `sv1`'s `Initialize` method gets fired.

Reply:
@tig since you do not want the `Initialized `event in the `View `it is no longer possible to trigger for each view. Also triggering the Initialized in `Toplevel `for each view will cause the same event to be fired several times unless `Toplevel `triggers it for each view but then the event must belong to the `View `class. If you want `Toplevel `to trigger the `Initialized `event for each view, I have to choose a method within the view where I can invoke it. Please give suggestions.

Reply:
> @tig since you do not want the `Initialized `event in the `View `it is no longer possible to trigger for each view. Also triggering the Initialized in `Toplevel `for each view will cause the same event to be fired several times unless `Toplevel `triggers it for each view but then the event must belong to the `View `class. If you want `Toplevel `to trigger the `Initialized `event for each view, I have to choose a method within the view where I can invoke it. Please give suggestions.

I previously suggested we invoke `Initialized` from `RunLoop`. I even provided a code sample that illustrated how it would work. I still do not understand why you rejected that idea.

Reply:
And about the Load event? Is to remove?

Reply:
https://github.com/migueldeicaza/gui.cs/issues/766#issuecomment-655735329

If you refer to this it is very confusing to me. I don't understand how you can trigger the `Initialized `event inside `RunLoop `for each view by moving to `Responder`. I think I'll give up on the idea anyway.